### PR TITLE
Handle authentication via JWT in the web app

### PR DESCRIPTION
**Pull request by morucci**
This change:

- Inspects the local storage to look for the JWT
- When authentication enabled and JWT absent or invalid then display the login page
- Login page displays a button to invite to login with the Authentication provider
- Add a timer that trigger the Login page display (delete the token from the storage) when the JWT expired
- Pass the JWT to API call if available in the store

Also some CI fixes are included.

### CODE REVIEW

File: haskell/src/Monocle/Config.hs [hunk 1]
```diff
@@ -295,15 +296,16 @@ getAuthProvider = do
   opClientSecret' <- lookupEnv "MONOCLE_OIDC_CLIENT_SECRET"
   opAppPublicURL' <- lookupEnv "MONOCLE_PUBLIC_URL"
   opUserClaim' <- lookupEnv "MONOCLE_OIDC_USER_CLAIM"
+  opEnforceAuth' <- lookupEnv "MONOCLE_ENFORCE_AUTH"
```
Comment (TristanCacqueray) [inline haskell/src/Monocle/Config.hs:299]: nit: adapt the value in one pass:
```suggestion
  opUserClaim <- fmap from <$> lookup "MONOCLE_OIDC_USER_CLAIM"
  opEnforceAuth <- maybe False (not . null) <$> lookupEnv "MONOCLE_ENFORCE_AUTH"
```

File: web/src/App.res [hunk 2]
```diff
@@ -245,43 +306,61 @@ module App = {
     let (showAbout, setShowAbout) = React.useState(_ => false)
     let (showLoginModal, setShowLoginModal) = React.useState(_ => false)
 
+    // Remove the token 60s before its expiration
```
Comment (TristanCacqueray) [inline web/src/App.res:309]: Would it be possible to start the setTimeout right when the token is read from the local storage? I wary of the `useEffect0` semantic...

File: web/src/components/Jwt.res [hunk 3]
```diff
@@ -0,0 +1,37 @@
+// Copyright (C) 2021 Monocle authors
```
Comment (TristanCacqueray) [inline web/src/components/Jwt.res:1]: nit: perhaps we should stop putting all the code in the components directory. How about we move this and the Store to a `web/src/lib` directory?

**Review by TristanCacqueray — APPROVED** [review]
Very well done!

File: web/src/App.res (hunk 2 above)
Comment (morucci) [reply web/src/App.res:309]: I tried but within the store's create function I'm not able to initialize the setTimeout cb with the dispatch action.

So if you know how to do this, please let me know if not let's move forward with this change.

**Review by TristanCacqueray — APPROVED** [review]

**Pull request merged**